File: app/src/main/java/com/kiss/www/kweather/Common/HttpHelper.java
package com.kiss.www.kweather.Common;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

/**
 * Created by Dre on 12/24/2017.
 */

public class HttpHelper {
    private static String stream = null;

    public HttpHelper() {

    }

    public String getHTTPData(String urlString)
    {

        try{
            URL url = new URL(urlString);
            HttpURLConnection urlConnection = (HttpURLConnection)url.openConnection();
            if(urlConnection.getResponseCode() == HttpURLConnection.HTTP_OK)
            {
                BufferedReader r = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));
                StringBuilder sb = new StringBuilder();
                String line;
                while((line = r.readLine())!=null)
                    sb.append(line);
                stream = sb.toString();
                urlConnection.disconnect();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        return stream;
    }

}

File: app/src/main/java/com/kiss/www/kweather/common/HttpHelper.kt
package com.kiss.www.kweather.common

import android.util.Log
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response
import java.lang.reflect.InvocationTargetException

class HttpHelper {
    suspend fun okHttpGet(urlString: String): Response? = withContext(Dispatchers.IO) {
        val request = Request.Builder()
                .url(urlString)
                .build()

        try {
            response = httpClient.newCall(request).execute()
            Log.i("OKHTTP", "RESPONSE -> ${response?.body.toString()}")
        } catch (ex: InvocationTargetException) {
            Log.e("OKHTTP", "Could not connect to target", ex)
        }
        response
    }

    companion object {
        private var stream: String? = null
        var response: Response? = null
        private val httpClient: OkHttpClient = OkHttpClient()
    }
}
